plugins/field-dependent-dropdown/README.md

```diff
@@ -0,0 +1,87 @@
+
+These changes are recorded properly in the undo history, and the fields can be [serialized and later deserialized](https://developers.google.com/blockly/guides/configure/web/serialization) while preserving their options, values, and validity. You can also create chains of dependent dropdowns that depend on other dependent dropdowns. 
+
+Note that the parent field must be attached to the block before the child field, and the child field will attach a validator function to the parent field to intercept changes to its value. If you want to add your own [custom field validator](https://developers.google.com/blockly/guides/create-custom-blocks/fields/validators) to the parent field, you need to use [the JavaScript API to define your block](https://developers.google.com/blockly/guides/create-custom-blocks/define-blocks#json_format_versus_javascript_api) and [pass your validator to the parent field's constructor](https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/dropdown#creating_a_dropdown_validator). If you try to set the parent's validator later, you'll overwrite the one added by this plugin.
+
+To create a dependent dropdown, you'll need to add this field type to a block definition, and add that block to your toolbox. See below for an example of defining a block that uses this field.
+
+### JSON
+
+```js
+import * as Blockly from 'blockly';
+import '@blockly/field-dependent-dropdown'; // Import with side effects.
+
+Blockly.defineBlocksWithJsonArray([
+  {
+    'type': 'dependent_dropdown_example',
+    'message0': 'Category %1 Animal %2',
+    'args0': [
+      {
+        'type': 'field_dropdown',
+        'name': 'ANIMAL_CATEGORY',
+        'options': [['Mammal', 'mammal'], ['Bird', 'bird']]
+      },
+      {
+        'type': 'field_dependent_dropdown',
+        'name': 'ANIMAL',
+        'parentName': 'ANIMAL_CATEGORY',
+        'optionMapping': {
+          'mammal': [['Dog', 'dog'], ['Cat', 'cat'], ['Hamster', 'hamster']],
+          'bird': [['Parakeet', 'parakeet'], ['Canary', 'canary']]
+        }
+      }
+    ]
+  }
+]);
+```
+
+### JavaScript
+
+```js
+import * as Blockly from 'blockly';
+import {FieldDependentDropdown} from '@blockly/field-dependent-dropdown';
+Blockly.Blocks['dependent_dropdown_example'] = {
+  init: function() {
+    const parentFieldName = 'ANIMAL_CATEGORY';
+    const childFieldName = 'ANIMAL';
+    const parentOptions = [['Mammal', 'mammal'], ['Bird', 'bird']];
+    const childOptions = {
```

Comment:
nit: could you call this `optionMapping` to be more consistent with the json api?